parsel/selector.py

```diff
@@ -346,6 +428,30 @@ def attrib(self):
         """
         return dict(self.root.attrib)
 
+    def cleaned(self, cleaner='html'):
+        """
+        Return a copy of a Selector, with underlying subtree cleaned.
+        Allowed values of ``cleaner`` argument:
+
+        * "html" (default) - use default ``lxml.html.clean.Cleaner``;
+        * "text" - clean with options tuned for text extraction: elements
+          like ``<script>`` and ``<style>`` are removed, cleaning options
+          are tuned for speed, assuming text extraction is the end goal;
+        * custom ``lxml.html.clean.Cleaner`` objects are also supported.
+        """
+        if isinstance(cleaner, six.string_types):
+            if cleaner not in {'html', 'text'}:
+                raise ValueError("cleaner must be 'html', 'text' or "
+                                 "an lxml.html.clean.Cleaner instance")
+        if cleaner == 'html':
+            cleaner = self._html_cleaner
+        elif cleaner == 'text':
+            cleaner = self._text_cleaner
```

Comment:
an alternative is make these attributes public, and ask users to pass them: ``sel.cleaned(sel.TEXT_CLEANER)`` instead of ``sel.cleaned('text')``.